cpp/oneapi/dal/algo/knn/test/batch.cpp

```diff
@@ -0,0 +1,339 @@
+        return knn::descriptor<Float, Method, knn::task::classification>(override_class_count,
+                                                                         override_neighbor_count);
+    }
+
+    static constexpr bool is_kd_tree = std::is_same_v<Method, knn::method::kd_tree>;
+    static constexpr bool is_brute_force = std::is_same_v<Method, knn::method::brute_force>;
+
+    bool not_available_on_device() {
+        return (get_policy().is_gpu() && is_kd_tree) || (get_policy().is_cpu() && is_brute_force);
+    }
+
+    te::table_id get_homogen_table_id() const {
+        return te::table_id::homogen<Float>();
+    }
+
+    Float classification(const table& train_data,
+                         const table& train_labels,
+                         const table& infer_data,
+                         const table& infer_labels,
+                         const std::int64_t n_classes,
+                         const std::int64_t n_neighbors = 1,
+                         const Float tolerance = Float(1.e-5)) {
+        SECTION("data shape is expected") {
+            REQUIRE(train_data.get_column_count() == infer_data.get_column_count());
+            REQUIRE(train_labels.get_column_count() == 1);
+            REQUIRE(infer_labels.get_column_count() == 1);
+            REQUIRE(infer_data.get_row_count() == infer_labels.get_row_count());
+            REQUIRE(train_data.get_row_count() == train_labels.get_row_count());
+        }
+
+        const auto knn_desc = this->get_descriptor(n_classes, n_neighbors);
+
+        auto train_result = this->train(knn_desc, train_data, train_labels);
+        auto train_model = train_result.get_model();
+        auto infer_result = this->infer(knn_desc, infer_data, train_model);
+        auto [prediction] = this->unpack_result(infer_result);
+
+        const auto score_table = te::accuracy_score<Float>(infer_labels, prediction, tolerance);
+        const auto score = row_accessor<const Float>(score_table).pull({ 0, -1 })[0];
+        return score;
+    }
+
+    void exact_nearest_indices_check(const table& train_data,
+                                     const table& infer_data,
+                                     const knn::infer_result<>& result) {
+        check_nans(result);
+
+        const auto [labels] = unpack_result(result);
+
+        const auto gtruth = naive_knn_search(train_data, infer_data);
+
+        SECTION("data shape is expected") {
+            REQUIRE(train_data.get_column_count() == infer_data.get_column_count());
+            REQUIRE(infer_data.get_row_count() == labels.get_row_count());
+            REQUIRE(labels.get_column_count() == 1);
+            REQUIRE(infer_data.get_row_count() == gtruth.get_row_count());
+            REQUIRE(train_data.get_row_count() == gtruth.get_column_count());
+        }
+
+        const auto m = infer_data.get_row_count();
+
+        const auto indices = naive_knn_search(train_data, infer_data);
+
+        for (std::int64_t j = 0; j < m; ++j) {
```

Comment:
```suggestion
        for (std::int64_t j = 0; j < infer_data.get_row_count(); ++j) {
```

Reply:
It's probably a little bit less performance